flit_core/flit_core/config.py

```diff
@@ -822,34 +824,96 @@ def isabs_ish(path):
     return os.path.isabs(path) or path.startswith(('/', '\\'))
 
 
-def normalize_license_expr(s: str):
-    """Validate & normalise an SPDX license expression
+def normalise_compound_license_expr(s: str) -> str:
+    """Validate and normalise a compund SPDX license expression.
 
-    For now this only handles simple expressions (referring to 1 license)
+    Per the specification, licence expression operators (AND, OR and WITH)
+    are matched case-sensitively. The WITH operator is not currently supported.
+
+    Spec: https://spdx.github.io/spdx-spec/v2.2.2/SPDX-license-expressions/
+    """
+    invalid_msg = "'{s}' is not a valid SPDX license expression: {reason}"
+    if not s or s.isspace():
+        raise ConfigError(f"The SPDX license expression must not be empty")
+
+    stack = 0
+    parts = []
+    try:
+        for part in filter(None, re.split(r' +|([()])', s)):
```

Comment:
If we wanted to preserve whitespace we can do so here -- [`re.split()`](https://docs.python.org/3/library/re.html#re.split) returns the text of all groups used in the capturing pattern.

Reply:
I'd prefer to keep it. Even though the PEP only mentions `case-normalized`, removing unnecessary whitespaces improves readability IMO. `packaging` does that as well.